lib/SILOptimizer/Utils/Existential.cpp

```diff
@@ -18,24 +18,20 @@
 
 using namespace swift;
 
-/// Determine the pattern for global_addr.
+/// Determine InitExistential from global_addr.
 /// %3 = global_addr @$P : $*SomeP
 /// %4 = init_existential_addr %3 : $*SomeP, $SomeC
 /// %5 = alloc_ref $SomeC
 /// store %5 to %4 : $*SomeC
 /// %8 = alloc_stack $SomeP
 /// copy_addr %3 to [initialization] %8 : $*SomeP
-/// %9 = open_existential_addr immutable_access %8 : $*SomeP to $*@opened SomeP
-SILValue swift::findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI,
-                                                  CopyAddrInst *CAI) {
-  assert(CAI->getSrc() == SILValue(GAI) &&
-         "Broken Assumption! Global Addr is not the source of the passed in "
-         "copy_addr?!");
-
+/// %10 = apply %9(%3) : $@convention(thin) (@in_guaranteed SomeP)
+static SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI,
+                                                  SILInstruction *Insn) {
```

Comment:
nit: you should probably comment that `Insn` is assumed to be a direct user of `GAI` and that a valid `init_existential_addr` value is returned only if this routine can prove that the value it initializes is the same value in memory at the given use point. Looking at this again, I don't understand why the return type is `SILValue` and not `InitExistentialAddrInst*`.